event_reporter/reporter/app_revision_helpers.go

```diff
@@ -0,0 +1,84 @@
+package reporter
+
+import (
+	"context"
+	"github.com/argoproj/argo-cd/v2/pkg/apiclient/application"
+	appv1 "github.com/argoproj/argo-cd/v2/pkg/apis/application/v1alpha1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+)
+
+type AppSyncRevisionsMetadata struct {
+	revisions     []*appv1.RevisionMetadata
+	syncRevisions []*appv1.RevisionMetadata
+}
+
+func getLatestAppHistoryId(a *appv1.Application) int64 {
+	var id int64
+	lastHistory := getLatestAppHistoryItem(a)
+
+	if lastHistory != nil {
+		id = lastHistory.ID
+	}
+
+	return id
+}
+
+func getLatestAppHistoryItem(a *appv1.Application) *appv1.RevisionHistory {
+	if a.Status.History != nil && len(a.Status.History) > 0 {
+		return &a.Status.History[len(a.Status.History)-1]
+	}
+
+	return nil
+}
+
+func getApplicationLatestRevision(a *appv1.Application) string {
+	revision := a.Status.Sync.Revision
+	lastHistory := getLatestAppHistoryItem(a)
+
+	if lastHistory != nil {
+		revision = lastHistory.Revision
+	}
+
+	return revision
+}
+
+func getOperationRevision(a *appv1.Application) string {
+	var revision string
+	if a != nil {
```

Comment:
lets change to if a == nil { return "" } , and everything outside the block, such code will be easier to read